### PR TITLE
Fix native mcm workflow following dynamic_one_shot refactor.

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-61844]

### CODE REVIEW

File: pennylane_lightning/lightning_qubit/_measurements.py [hunk 1]
```diff
@@ -285,27 +287,37 @@ def measure_final_state(self, circuit: QuantumScript) -> Result:
...
+            measurements (List[Union[SampleMeasurement, ClassicalShadowMP, ShadowExpvalMP]]):
                 The sample measurements to perform
             shots (Shots): The number of samples to take
+            mid_measurements (None, dict): Dictionary of mid-circuit measurements
 
         Returns:
             List[TensorLike[Any]]: Sample measurement results
         """
+        # last N measurements are sampling MCMs in ``dynamic_one_shot`` execution mode
+        mps = measurements[0 : -len(mid_measurements)] if mid_measurements else measurements
```
Comment (maliasadi) [inline pennylane_lightning/lightning_qubit/_measurements.py:309]: Should we check this or this is always preserved from the callee? 
```suggestion
        assert(len(mid_measurements) <= len(measurements))
        mps = measurements[0 : -len(mid_measurements)] if mid_measurements else measurements
```

File: pennylane_lightning/lightning_qubit/_measurements.py [hunk 2]
```diff
@@ -285,27 +287,37 @@ def measure_final_state(self, circuit: QuantumScript) -> Result:
...
+            any(v == -1 for v in mid_measurements.values()) if mid_measurements else False
+        )
 
         groups, indices = _group_measurements(mps)
 
         all_res = []
         for group in groups:
+            if skip_measure:
+                all_res.extend([None] * len(group))
+                continue
```
Comment (maliasadi) [inline pennylane_lightning/lightning_qubit/_measurements.py:320]: what is this logic for?

**Review by maliasadi — COMMENTED** [review]
Thanks @vincentmr! :eagle:

File: pennylane_lightning/lightning_qubit/_measurements.py (hunk 2 above)
Comment (vincentmr) [reply pennylane_lightning/lightning_qubit/_measurements.py:320]: If any of the MCM values is `-1`, the state might not be normalized which may lead to errors in measurement processes. We therefore just skip it.

File: pennylane_lightning/lightning_qubit/_measurements.py (hunk 1 above)
Comment (vincentmr) [reply pennylane_lightning/lightning_qubit/_measurements.py:309]: Will close as per our discussion.

**Review by maliasadi — APPROVED** [review]

**Pull request merged**